OneCommunityGlobal/HighestGoodNetworkApp
# Kaixiang edit and delete role

Backend PR: https://github.com/OneCommunityGlobal/HGNRest/pull/269
This PR added new functionalities for permission management 1. edit role name 2. delete role

![Capture](https://user-images.githubusercontent.com/98206107/212433973-d5503201-bab6-40eb-9ed6-c725a6e4731f.JPG)

### 🆕 What's New
1. Added new buttons for users to edit role names and delete roles.
2. A modal will popup when click **Edit Icon** which is beside role name.
3. A confirmation modal will popup when click **Delete Role** button.
4. Save reminder will show up when you make any changes on role permission page, and disappear when click save button.
![screenshot](https://user-images.githubusercontent.com/98206107/212434693-2f569148-59d1-4f16-bb60-4e30adcd295f.JPG)

### 👀 How to Test (UPDATED on 1/25)
1. Login with an **OWNER** account, go to Other Links -> Permissions Management.
2. Create a new test role by click **Add New Role** button.
3. Go to role permission page by click the role you just created.
4. Try edit role name and click save button see if it's saved correctly. 
5. Delete the role that you just created.
6. Keep an eye on the save reminder, see if it's pop-up and disappear properly.
7. **NEW:** Change your account role to ADMIN role on user profile change. You should have different view on permission page and you are not allowed to do any modifications for user role because all those buttons are disappeared.

### New changes made on 1/25
1. After I confirmed with Jae, any add/update/delete actions with user role **ONLY works for OWNER role account**.
2. I hide all the buttons for users who are not OWNER. Add/Delete buttons are disabled for users who are now Owner, but at least they can see which permissions are included which are not.
3. So, you should have different view when you log in with OWNER account and ADMIN account.
![hide_buttons_when_user_is_not_owner_1](https://user-images.githubusercontent.com/98206107/214928267-5eed0231-affb-4b89-95a3-76766d561a05.JPG)
![hide_buttons_when_user_is_not_owner_2](https://user-images.githubusercontent.com/98206107/214928269-e3e7379f-2d5a-4df2-b833-9311579343a3.JPG)
